test/common/Promise.spec.ts

```diff
@@ -9,12 +9,22 @@
 import {ifEnvSupports} from '../test-util';
 declare const global: any;
 
+let useZoneAwarePromise: boolean = true; 
+try {
+  Zone.assertZonePatched();
+} catch (error) {
+  useZoneAwarePromise = false;
+}
+
 class MicroTaskQueueZoneSpec implements ZoneSpec {
   name: string = 'MicroTaskQueue';
   queue: MicroTask[] = [];
   properties = {queue: this.queue, flush: this.flush.bind(this)};
 
   flush() {
+    if (!useZoneAwarePromise) {
```

Comment:
Why do we need to disable the tests? I would imagine that the tests should still work even with async_hook?